{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "45c4fb3d-2872-4ef9-8bb2-e8b7c60b1fb7",
   "metadata": {},
   "source": [
    "# Creating the sample_collab.db SQLite database"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "95591431-b1c3-4808-ada3-7eb9c986cfd9",
   "metadata": {},
   "source": [
    "To be run a single time to set prefered default sqlite3 output mode (here \"box\" type output, with column headers):"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "74b227b4-9dd5-4258-b341-db4a8200f189",
   "metadata": {},
   "outputs": [],
   "source": [
    "echo -e '.mode box\\n.headers on\\n' > ~/.sqliterc"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cc10261f-e40b-423c-89a6-57f838276b2f",
   "metadata": {},
   "source": [
    "## Create the sqlite DB from original raw data files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fb45825a-d47a-4f9a-9374-08a659dd84ce",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Change the path below to match your folder's path:\n",
    "cd /shared/projects/form_2022_19/==your_firstname==/sars2copath\n",
    "git status\n",
    "# if this does NOT return \"nothing to commit, working directory clean\", then you must first stage & commit your current local branch modifications:\n",
    "# git add .\n",
    "# git commit -m \"some message\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "c3710247-77ea-4fce-a220-471136c68efa",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "M\tanalyses/sample_collab.db\n",
      "Switched to branch 'sample_collab_db_creation'\n",
      "total 136\n",
      "-rw-r-----+ 1 phingamp phingamp 126976 May 20 00:28 sample_collab.db\n",
      "-rw-rw----+ 1 phingamp phingamp   5914 May 20 00:34 sample_collab_db_schema.sql\n",
      "total 740\n",
      "-rw-rw----+ 1 phingamp phingamp 214743 May 13 20:48 SraRunTable.csv\n",
      "-rw-rw----+ 1 phingamp phingamp  31441 May 19 13:39 sample_overviews 5.csv\n",
      "-rw-rw----+ 1 phingamp phingamp  31719 May 19 13:39 sample_overviews 3.csv\n",
      "-rw-rw----+ 1 phingamp phingamp  15884 May 19 13:39 ct_values_covid19_metagenomics.xlsx\n",
      "-rw-rw----+ 1 phingamp phingamp 447419 May 19 13:39 filereport_read_run_PRJEB47870.tsv\n"
     ]
    }
   ],
   "source": [
    "git branch\n",
    "git branch sample_collab_db_creation\n",
    "git checkout sample_collab_db_creation\n",
    "cd ./analyses\n",
    "ls -ltr\n",
    "ls -ltr ../data"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "064da25b-20a8-4b39-bf00-ba8655ca7163",
   "metadata": {},
   "source": [
    "## Don't forget at the end of these sandbox explorations to quit this git branch and return to your real development branch (see instructions at the end)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "b2022812-bdb9-463f-b600-310909084766",
   "metadata": {},
   "outputs": [],
   "source": [
    "[ -f sample_collab.db ] && rm sample_collab.db\n",
    "sqlite3 -batch  sample_collab.db < sample_collab_db_schema.sql"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "cd86ca84-d133-4342-9017-19d5f05885fb",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "â”Œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”\n",
      "â”‚ count(*) â”‚\n",
      "â”œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¤\n",
      "â”‚ 250      â”‚\n",
      "â””â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”˜\n"
     ]
    }
   ],
   "source": [
    "sed -E \"s/,P158cDNA_L001,/,P158_cDNA_L001,/\" ../data/SraRunTable.csv | sed -E \"s/,P011_S39_L001,/,P011_DNA_S39_L001,/\" > ../data/SraRunTable_mended.csv\n",
    "sqlite3 -batch sample_collab.db \".import --csv --skip 1 ../data/SraRunTable_mended.csv SraRunTable\"\n",
    "#sqlite3 -batch sample_collab.db \".schema SraRunTable\"\n",
    "sqlite3 -batch sample_collab.db \"select count(*) from SraRunTable\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "e6064b09-a978-4fd0-b04a-395577d54725",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "â”Œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”\n",
      "â”‚ count(*) â”‚\n",
      "â”œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¤\n",
      "â”‚ 250      â”‚\n",
      "â””â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”˜\n"
     ]
    }
   ],
   "source": [
    "sqlite3 -batch -tabs sample_collab.db \".import --skip 1 ../data/filereport_read_run_PRJEB47870.tsv ENARunTable\"\n",
    "#sqlite3 -batch sample_collab.db \".schema ENARunTable\"\n",
    "sqlite3 -batch sample_collab.db \"select count(*) from ENARunTable\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "e568a243-2ccc-493e-ac72-813429d80eb6",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "251 ../data/sample_overviews_all.csv\n",
      "â”Œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”\n",
      "â”‚ count(*) â”‚\n",
      "â”œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¤\n",
      "â”‚ 250      â”‚\n",
      "â””â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”˜\n"
     ]
    }
   ],
   "source": [
    "cat \"../data/sample_overviews 5.csv\" > ../data/sample_overviews_all.csv\n",
    "tail -n +2 \"../data/sample_overviews 3.csv\" >> ../data/sample_overviews_all.csv\n",
    "wc -l ../data/sample_overviews_all.csv\n",
    "sed -E \"s/^P158cDNA_L001,/P158_cDNA_L001,/\" ../data/sample_overviews_all.csv | sed -E \"s/^P011_S39_L001,/P011_DNA_S39_L001,/\" > ../data/sample_overviews_all_mended.csv\n",
    "tail -n +2 ../data/sample_overviews_all_mended.csv | cut -f1 -d\",\" | sed -E \"s/^(\\S+)_(DNA|cDNA|CDNA|RNA|dna).*$/,\\1/\" | sed -E \"s/P07_gg/P07/\" | sed -E \"s/^,P0+/,P/\" > ../data/sample_overviews_all_mended_patient_code.csv\n",
    "echo \",patient_code\" | cat - ../data/sample_overviews_all_mended_patient_code.csv > ../data/sample_overviews_all_mended_patient_code_with_header.csv\n",
    "paste ../data/sample_overviews_all_mended.csv ../data/sample_overviews_all_mended_patient_code_with_header.csv > ../data/sample_overviews_all_mended_patient_code_with_header_pasted.csv\n",
    "sqlite3 -batch -csv sample_collab.db \".import --csv --skip 1 ../data/sample_overviews_all_mended_patient_code_with_header_pasted.csv SampleOverviews\"\n",
    "#sqlite3 -batch sample_collab.db \".schema SampleOverviews\"\n",
    "sqlite3 -batch sample_collab.db \"select count(*) from SampleOverviews\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "b2272754-3be3-4fb5-8504-bcb53c6d501e",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "module load csvkit\n",
    "in2csv ../data/ct_values_covid19_metagenomics.xlsx > ../data/ct_values_covid19_metagenomics.csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "c5ec4347-f5e4-4de3-b8d3-99adb5e09c50",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "â”Œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”\n",
      "â”‚ count(*) â”‚\n",
      "â”œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¤\n",
      "â”‚ 379      â”‚\n",
      "â””â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”˜\n"
     ]
    }
   ],
   "source": [
    "sqlite3 -batch sample_collab.db \".import --csv --skip 1 ../data/ct_values_covid19_metagenomics.csv CtValues\"\n",
    "#sqlite3 -batch sample_collab.db \".schema CtValues\"\n",
    "sqlite3 -batch sample_collab.db \"select count(*) from CtValues\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "12092a94-7400-4cd4-9c67-12a0779819bc",
   "metadata": {},
   "source": [
    "## ğŸ˜‰ Warm up with an easy join between TWO tables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "d1337f93-ffd8-4123-873a-09ec06d773a5",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "â”Œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”\n",
      "â”‚ count(*) â”‚\n",
      "â”œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¤\n",
      "â”‚ 250      â”‚\n",
      "â””â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”˜\n"
     ]
    }
   ],
   "source": [
    "sqlite3 -batch sample_collab.db \\\n",
    "\"select count(*) \\\n",
    "from SraRunTable sra inner join SampleOverviews spl on sra.host_subject_id=spl.sample_name;\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3adab714-6d21-4e87-b503-d505d073745d",
   "metadata": {},
   "source": [
    "Quality control: Check that sample names are coherent between NCBI SRA and EBI ENA:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "af96c487-058f-47bb-ba00-c46e103f2cee",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "â”Œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”\n",
      "â”‚ count(*) â”‚\n",
      "â”œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¤\n",
      "â”‚ 250      â”‚\n",
      "â””â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”˜\n",
      "â”Œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”\n",
      "â”‚ count(*) â”‚\n",
      "â”œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¤\n",
      "â”‚ 250      â”‚\n",
      "â””â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”˜\n"
     ]
    }
   ],
   "source": [
    "sqlite3 -batch sample_collab.db \\\n",
    "\"select count(*) from  SampleOverviews spl left join SraRunTable sra on sra.host_subject_id=spl.sample_name;\"\n",
    "sqlite3 -batch sample_collab.db \\\n",
    "\"select count(*) from SraRunTable sra left join SampleOverviews spl on sra.host_subject_id=spl.sample_name;\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e6fafbb1-6af4-4327-8332-36fead8cf44d",
   "metadata": {},
   "source": [
    "## ğŸ¤¨ Let's go for a DOUBLE join between THREE tables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "f404e30f-7b82-46c9-86a3-e64e1ac23752",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "â”Œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”\n",
      "â”‚ count(*) â”‚\n",
      "â”œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¤\n",
      "â”‚ 250      â”‚\n",
      "â””â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”˜\n"
     ]
    }
   ],
   "source": [
    "sqlite3 -batch sample_collab.db \"select count(*) \\\n",
    "from (SraRunTable sra inner join SampleOverviews spl on sra.host_subject_id=spl.sample_name) \\\n",
    "inner join ENARunTable ena on sra.Run=ena.run_accession;\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5c04f558-115e-480d-ad79-982ee83702c8",
   "metadata": {},
   "source": [
    "## ğŸ˜² Take a deep breath : now for the TRIPLE join of the FOUR tables:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "76d0801b-222f-4687-b4f9-bb15045c29d5",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "â”Œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”\n",
      "â”‚ count(*) â”‚\n",
      "â”œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¤\n",
      "â”‚ 250      â”‚\n",
      "â””â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”˜\n"
     ]
    }
   ],
   "source": [
    "sqlite3 -batch sample_collab.db \\\n",
    "\"select count(*) \\\n",
    "from ((SraRunTable sra inner join SampleOverviews spl on sra.host_subject_id=spl.sample_name) \\\n",
    "inner join ENARunTable ena on sra.Run=ena.run_accession) \\\n",
    "left join CtValues ct on patient_code=ct.Blinded_number;\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "01189d4b-6dfd-4d79-a863-8aeca94a48a0",
   "metadata": {},
   "source": [
    "## Create a couple of views to simplify usage of the sample annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "8c01c6cb-54e5-4c24-a506-150a6333b975",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "sqlite3 -batch sample_collab.db \\\n",
    "\"create view sample_annot_all as select * \\\n",
    "from ((SraRunTable sra inner join SampleOverviews spl on sra.host_subject_id=spl.sample_name) \\\n",
    "inner join ENARunTable ena on sra.Run=ena.run_accession) \\\n",
    "left join CtValues ct on patient_code=ct.Blinded_number;\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "42ef8e47-e41d-481d-b5b1-c87a940d0a79",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "sqlite3 -batch sample_collab.db \\\n",
    "\"create view sample_annot as select run_accession,host_subject_id,patient_code,nucleotide_type as nuc,host_body_site,\\\n",
    "    host_disease_status,miscellaneous_parameter,Ct,total_reads,read_count,base_count  \\\n",
    "from ((SraRunTable sra inner join SampleOverviews spl on sra.host_subject_id=spl.sample_name) \\\n",
    "inner join ENARunTable ena on sra.Run=ena.run_accession) \\\n",
    "left join CtValues ct on patient_code=ct.Blinded_number;\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f1198e66-a997-4d53-923c-c31d55c9325f",
   "metadata": {},
   "source": [
    "### From now on we can just use the \"sample_annot\" view as if it was an ordinary table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "3ce0699f-4d7d-45e8-b62b-d921c1f8ea64",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "â”Œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”\n",
      "â”‚ run_accession â”‚  host_subject_id  â”‚ patient_code â”‚ nuc â”‚ host_body_site â”‚ host_disease_status â”‚  miscellaneous_parameter   â”‚  Ct   â”‚ total_reads â”‚ read_count â”‚ base_count â”‚\n",
      "â”œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¤\n",
      "â”‚ ERR6913102    â”‚ P011_DNA_S39_L001 â”‚ P11          â”‚ DNA â”‚ Nasopharyngeal â”‚ SARS-CoV-2 positive â”‚ Symptomatic-DNA-sequenced  â”‚ 23.59 â”‚ 27630988    â”‚ 102524     â”‚ 13193813   â”‚\n",
      "â”‚ ERR6913103    â”‚ P019_DNA_S74_L001 â”‚ P19          â”‚ DNA â”‚ Nasopharyngeal â”‚ SARS-CoV-2 positive â”‚ Symptomatic-DNA-sequenced  â”‚ 18.81 â”‚ 21364260    â”‚ 385740     â”‚ 49060845   â”‚\n",
      "â”‚ ERR6913108    â”‚ P069_DNA_S27_L001 â”‚ P69          â”‚ DNA â”‚ Nasopharyngeal â”‚ SARS-CoV-2 positive â”‚ Symptomatic-DNA-sequenced  â”‚ 18.42 â”‚ 23356096    â”‚ 226280     â”‚ 29240147   â”‚\n",
      "â”‚ ERR6913111    â”‚ P090_DNA_S25_L001 â”‚ P90          â”‚ DNA â”‚ Nasopharyngeal â”‚ SARS-CoV-2 positive â”‚ Symptomatic-DNA-sequenced  â”‚ 26.98 â”‚ 27646936    â”‚ 33240      â”‚ 4397612    â”‚\n",
      "â”‚ ERR6913115    â”‚ P267_DNA_S9_L001  â”‚ P267         â”‚ DNA â”‚ Nasopharyngeal â”‚ SARS-CoV-2 positive â”‚ Symptomatic-DNA-sequenced  â”‚ 19.63 â”‚ 18306476    â”‚ 33254      â”‚ 4259170    â”‚\n",
      "â”‚ ERR6913116    â”‚ P269_DNA_S3_L001  â”‚ P269         â”‚ DNA â”‚ Nasopharyngeal â”‚ SARS-CoV-2 positive â”‚ Symptomatic-DNA-sequenced  â”‚ 23.43 â”‚ 21061004    â”‚ 43790      â”‚ 5716941    â”‚\n",
      "â”‚ ERR6913117    â”‚ P272_DNA_S40_L001 â”‚ P272         â”‚ DNA â”‚ Nasopharyngeal â”‚ SARS-CoV-2 positive â”‚ Symptomatic-DNA-sequenced  â”‚ 21.34 â”‚ 24205196    â”‚ 397186     â”‚ 50556736   â”‚\n",
      "â”‚ ERR6913118    â”‚ P273_DNA_L001     â”‚ P273         â”‚ DNA â”‚ Nasopharyngeal â”‚ SARS-CoV-2 positive â”‚ Symptomatic-DNA-sequenced  â”‚ 17.81 â”‚ 16700384    â”‚ 140270     â”‚ 16091403   â”‚\n",
      "â”‚ ERR6913136    â”‚ P27_DNA_L001_2    â”‚ P27          â”‚ DNA â”‚ Nasopharyngeal â”‚ SARS-CoV-2 positive â”‚ Asymptomatic-DNA-sequenced â”‚ 33.4  â”‚ 20026148    â”‚ 2239760    â”‚ 318332989  â”‚\n",
      "â”‚ ERR6913147    â”‚ P38_DNA_L001_2    â”‚ P38          â”‚ DNA â”‚ Nasopharyngeal â”‚ SARS-CoV-2 positive â”‚ Asymptomatic-DNA-sequenced â”‚ 29.33 â”‚ 18680552    â”‚ 1994778    â”‚ 274525044  â”‚\n",
      "â”‚ ERR6913154    â”‚ P388_DNA_L001     â”‚ P388         â”‚ DNA â”‚ Nasopharyngeal â”‚ SARS-CoV-2 negative â”‚ Symptomatic-DNA-sequenced  â”‚ 0.0   â”‚ 18749066    â”‚ 39244      â”‚ 5260372    â”‚\n",
      "â”‚ ERR6913157    â”‚ P405_DNA_S13_L001 â”‚ P405         â”‚ DNA â”‚ Nasopharyngeal â”‚ SARS-CoV-2 negative â”‚ Symptomatic-DNA-sequenced  â”‚ 0.0   â”‚ 29058038    â”‚ 171728     â”‚ 22320314   â”‚\n",
      "â”‚ ERR6913161    â”‚ P410_DNA_S61_L001 â”‚ P410         â”‚ DNA â”‚ Nasopharyngeal â”‚ SARS-CoV-2 negative â”‚ Symptomatic-DNA-sequenced  â”‚ 0.0   â”‚ 19191850    â”‚ 84520      â”‚ 11211869   â”‚\n",
      "â”‚ ERR6913165    â”‚ P419_DNA_S51_L001 â”‚ P419         â”‚ DNA â”‚ Nasopharyngeal â”‚ SARS-CoV-2 negative â”‚ Symptomatic-DNA-sequenced  â”‚ 0.0   â”‚ 20304462    â”‚ 205606     â”‚ 26698288   â”‚\n",
      "â”‚ ERR6913170    â”‚ P428_DNA_S45_L001 â”‚ P428         â”‚ DNA â”‚ Nasopharyngeal â”‚ SARS-CoV-2 negative â”‚ Symptomatic-DNA-sequenced  â”‚ 0.0   â”‚ 23713614    â”‚ 16132      â”‚ 2129809    â”‚\n",
      "â”‚ ERR6913173    â”‚ P435_DNA_S33_L001 â”‚ P435         â”‚ DNA â”‚ Nasopharyngeal â”‚ SARS-CoV-2 negative â”‚ Symptomatic-DNA-sequenced  â”‚ 0.0   â”‚ 21165410    â”‚ 27156      â”‚ 3357535    â”‚\n",
      "â”‚ ERR6913174    â”‚ NP_A_DNA_L001     â”‚ NP_A         â”‚ DNA â”‚ Nasopharyngeal â”‚ SARS-CoV-2 negative â”‚ Asymptomatic-DNA-sequenced â”‚       â”‚ 26613104    â”‚ 191868     â”‚ 21290514   â”‚\n",
      "â”‚ ERR6913175    â”‚ NP_B_DNA_L001     â”‚ NP_B         â”‚ DNA â”‚ Nasopharyngeal â”‚ SARS-CoV-2 negative â”‚ Asymptomatic-DNA-sequenced â”‚       â”‚ 19924574    â”‚ 399904     â”‚ 42664989   â”‚\n",
      "â”‚ ERR6913179    â”‚ P01_DNA_L001_2    â”‚ P1           â”‚ DNA â”‚ Nasopharyngeal â”‚ SARS-CoV-2 negative â”‚ Asymptomatic-DNA-sequenced â”‚ 27.65 â”‚ 17377298    â”‚ 1769196    â”‚ 248757369  â”‚\n",
      "â”‚ ERR6913189    â”‚ P11_DNA_L001_2    â”‚ P11          â”‚ DNA â”‚ Nasopharyngeal â”‚ SARS-CoV-2 negative â”‚ Asymptomatic-DNA-sequenced â”‚ 23.59 â”‚ 20686584    â”‚ 2225632    â”‚ 314971171  â”‚\n",
      "â””â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”˜\n"
     ]
    }
   ],
   "source": [
    "sqlite3 -batch sample_collab.db \\\n",
    "\"select * from sample_annot limit 20;\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "4a971d6b-41bf-48d4-a72b-537dfafea0cf",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "â”Œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”\n",
      "â”‚ run_accession â”‚                                                                                    submitted_ftp                                                                                    â”‚\n",
      "â”œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¤\n",
      "â”‚ ERR6913102    â”‚ ftp.sra.ebi.ac.uk/vol1/run/ERR691/ERR6913102/P011_S39_L001_1_68215_reads_nh_R1.fastq.gz;ftp.sra.ebi.ac.uk/vol1/run/ERR691/ERR6913102/P011_S39_L001_1_68215_reads_nh_R2.fastq.gz     â”‚\n",
      "â”‚ ERR6913103    â”‚ ftp.sra.ebi.ac.uk/vol1/run/ERR691/ERR6913103/P019_DNA_S74_L001_68214_reads_nh_R1.fastq.gz;ftp.sra.ebi.ac.uk/vol1/run/ERR691/ERR6913103/P019_DNA_S74_L001_68214_reads_nh_R2.fastq.gz â”‚\n",
      "â”‚ ERR6913108    â”‚ ftp.sra.ebi.ac.uk/vol1/run/ERR691/ERR6913108/P069_DNA_S27_L001_68232_reads_nh_R1.fastq.gz;ftp.sra.ebi.ac.uk/vol1/run/ERR691/ERR6913108/P069_DNA_S27_L001_68232_reads_nh_R2.fastq.gz â”‚\n",
      "â”‚ ERR6913111    â”‚ ftp.sra.ebi.ac.uk/vol1/run/ERR691/ERR6913111/P090_DNA_S25_L001_68234_reads_nh_R1.fastq.gz;ftp.sra.ebi.ac.uk/vol1/run/ERR691/ERR6913111/P090_DNA_S25_L001_68234_reads_nh_R2.fastq.gz â”‚\n",
      "â”‚ ERR6913115    â”‚ ftp.sra.ebi.ac.uk/vol1/run/ERR691/ERR6913115/P267_DNA_S9_L001_68207_reads_nh_R1.fastq.gz;ftp.sra.ebi.ac.uk/vol1/run/ERR691/ERR6913115/P267_DNA_S9_L001_68207_reads_nh_R2.fastq.gz   â”‚\n",
      "â”‚ ERR6913116    â”‚ ftp.sra.ebi.ac.uk/vol1/run/ERR691/ERR6913116/P269_DNA_S3_L001_68206_reads_nh_R1.fastq.gz;ftp.sra.ebi.ac.uk/vol1/run/ERR691/ERR6913116/P269_DNA_S3_L001_68206_reads_nh_R2.fastq.gz   â”‚\n",
      "â”‚ ERR6913117    â”‚ ftp.sra.ebi.ac.uk/vol1/run/ERR691/ERR6913117/P272_DNA_S40_L001_68205_reads_nh_R1.fastq.gz;ftp.sra.ebi.ac.uk/vol1/run/ERR691/ERR6913117/P272_DNA_S40_L001_68205_reads_nh_R2.fastq.gz â”‚\n",
      "â”‚ ERR6913118    â”‚ ftp.sra.ebi.ac.uk/vol1/run/ERR691/ERR6913118/P273_DNA_L001_74640_reads_nh_R1.fastq.gz;ftp.sra.ebi.ac.uk/vol1/run/ERR691/ERR6913118/P273_DNA_L001_74640_reads_nh_R2.fastq.gz         â”‚\n",
      "â”‚ ERR6913136    â”‚ ftp.sra.ebi.ac.uk/vol1/run/ERR691/ERR6913136/P27_DNA_L001_2_76130_reads_nh_R1.fastq.gz;ftp.sra.ebi.ac.uk/vol1/run/ERR691/ERR6913136/P27_DNA_L001_2_76130_reads_nh_R2.fastq.gz       â”‚\n",
      "â”‚ ERR6913147    â”‚ ftp.sra.ebi.ac.uk/vol1/run/ERR691/ERR6913147/P38_DNA_L001_2_76151_reads_nh_R1.fastq.gz;ftp.sra.ebi.ac.uk/vol1/run/ERR691/ERR6913147/P38_DNA_L001_2_76151_reads_nh_R2.fastq.gz       â”‚\n",
      "â”‚ ERR6913154    â”‚ ftp.sra.ebi.ac.uk/vol1/run/ERR691/ERR6913154/P388_DNA_L001_74605_reads_nh_R1.fastq.gz;ftp.sra.ebi.ac.uk/vol1/run/ERR691/ERR6913154/P388_DNA_L001_74605_reads_nh_R2.fastq.gz         â”‚\n",
      "â”‚ ERR6913157    â”‚ ftp.sra.ebi.ac.uk/vol1/run/ERR691/ERR6913157/P405_DNA_S13_L001_68236_reads_nh_R1.fastq.gz;ftp.sra.ebi.ac.uk/vol1/run/ERR691/ERR6913157/P405_DNA_S13_L001_68236_reads_nh_R2.fastq.gz â”‚\n",
      "â”‚ ERR6913161    â”‚ ftp.sra.ebi.ac.uk/vol1/run/ERR691/ERR6913161/P410_DNA_S61_L001_68228_reads_nh_R1.fastq.gz;ftp.sra.ebi.ac.uk/vol1/run/ERR691/ERR6913161/P410_DNA_S61_L001_68228_reads_nh_R2.fastq.gz â”‚\n",
      "â”‚ ERR6913165    â”‚ ftp.sra.ebi.ac.uk/vol1/run/ERR691/ERR6913165/P419_DNA_S51_L001_68224_reads_nh_R1.fastq.gz;ftp.sra.ebi.ac.uk/vol1/run/ERR691/ERR6913165/P419_DNA_S51_L001_68224_reads_nh_R2.fastq.gz â”‚\n",
      "â”‚ ERR6913170    â”‚ ftp.sra.ebi.ac.uk/vol1/run/ERR691/ERR6913170/P428_DNA_S45_L001_68219_reads_nh_R1.fastq.gz;ftp.sra.ebi.ac.uk/vol1/run/ERR691/ERR6913170/P428_DNA_S45_L001_68219_reads_nh_R2.fastq.gz â”‚\n",
      "â”‚ ERR6913173    â”‚ ftp.sra.ebi.ac.uk/vol1/run/ERR691/ERR6913173/P435_DNA_S33_L001_68216_reads_nh_R1.fastq.gz;ftp.sra.ebi.ac.uk/vol1/run/ERR691/ERR6913173/P435_DNA_S33_L001_68216_reads_nh_R2.fastq.gz â”‚\n",
      "â”‚ ERR6913174    â”‚ ftp.sra.ebi.ac.uk/vol1/run/ERR691/ERR6913174/NP_A_DNA_L001_76229_reads_nh_R1.fastq.gz;ftp.sra.ebi.ac.uk/vol1/run/ERR691/ERR6913174/NP_A_DNA_L001_76229_reads_nh_R2.fastq.gz         â”‚\n",
      "â”‚ ERR6913175    â”‚ ftp.sra.ebi.ac.uk/vol1/run/ERR691/ERR6913175/NP_B_DNA_L001_74600_reads_nh_R1.fastq.gz;ftp.sra.ebi.ac.uk/vol1/run/ERR691/ERR6913175/NP_B_DNA_L001_74600_reads_nh_R2.fastq.gz         â”‚\n",
      "â”‚ ERR6913179    â”‚ ftp.sra.ebi.ac.uk/vol1/run/ERR691/ERR6913179/P01_DNA_L001_2_76153_reads_nh_R1.fastq.gz;ftp.sra.ebi.ac.uk/vol1/run/ERR691/ERR6913179/P01_DNA_L001_2_76153_reads_nh_R2.fastq.gz       â”‚\n",
      "â”‚ ERR6913189    â”‚ ftp.sra.ebi.ac.uk/vol1/run/ERR691/ERR6913189/P11_DNA_L001_2_76173_reads_nh_R1.fastq.gz;ftp.sra.ebi.ac.uk/vol1/run/ERR691/ERR6913189/P11_DNA_L001_2_76173_reads_nh_R2.fastq.gz       â”‚\n",
      "â””â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”˜\n"
     ]
    }
   ],
   "source": [
    "sqlite3 -batch sample_collab.db \\\n",
    "\"select run_accession,submitted_ftp from sample_annot_all limit 20;\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8653f508-3c38-49e9-b979-c77b54e036e6",
   "metadata": {},
   "source": [
    "## More quality controling"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "542d389e-e643-4bb2-b23a-145c6fd65671",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "â”Œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”\n",
      "â”‚ count(distinct(patient_code)) â”‚\n",
      "â”œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¤\n",
      "â”‚ 120                           â”‚\n",
      "â””â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”˜\n"
     ]
    }
   ],
   "source": [
    "sqlite3 -batch sample_collab.db \\\n",
    "\"select count(distinct(patient_code)) \\\n",
    "from ((SraRunTable sra inner join SampleOverviews spl on sra.host_subject_id=spl.sample_name) \\\n",
    "inner join ENARunTable ena on sra.Run=ena.run_accession) \\\n",
    "left join CtValues ct on patient_code=ct.Blinded_number;\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "1ab7004e-ada2-4128-bfa6-6cd8fc6d807b",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "â”Œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”\n",
      "â”‚ count(distinct(patient_code)) â”‚\n",
      "â”œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¤\n",
      "â”‚ 104                           â”‚\n",
      "â””â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”˜\n"
     ]
    }
   ],
   "source": [
    "sqlite3 -batch sample_collab.db \\\n",
    "\"select count(distinct(patient_code)) \\\n",
    "from ((SraRunTable sra inner join SampleOverviews spl on sra.host_subject_id=spl.sample_name) \\\n",
    "inner join ENARunTable ena on sra.Run=ena.run_accession) \\\n",
    "left join CtValues ct on patient_code=ct.Blinded_number \\\n",
    "where patient_code like 'P%';\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "36c2805d-9015-44ee-ba34-fc249fb26215",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "â”Œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”\n",
      "â”‚ patient_code â”‚  host_subject_id  â”‚ nb â”‚\n",
      "â”œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”¤\n",
      "â”‚ P11          â”‚ P011_DNA_S39_L001 â”‚ 4  â”‚\n",
      "â”‚ P19          â”‚ P019_DNA_S74_L001 â”‚ 4  â”‚\n",
      "â”‚ P22          â”‚ P022_DNA_L001     â”‚ 4  â”‚\n",
      "â”‚ P35          â”‚ P035_DNA_S21_L001 â”‚ 4  â”‚\n",
      "â”‚ P4           â”‚ P004_DNA_L001     â”‚ 4  â”‚\n",
      "â”‚ NP_A         â”‚ NP_A_DNA_L001     â”‚ 2  â”‚\n",
      "â”‚ NP_B         â”‚ NP_B_DNA_L001     â”‚ 2  â”‚\n",
      "â”‚ NP_C         â”‚ NP_C_DNA_L001     â”‚ 2  â”‚\n",
      "â”‚ NP_D         â”‚ NP_D_DNA_L001     â”‚ 2  â”‚\n",
      "â”‚ NP_E         â”‚ NP_E_DNA_L001     â”‚ 2  â”‚\n",
      "â”‚ P1           â”‚ P01_DNA_L001_2    â”‚ 2  â”‚\n",
      "â”‚ P10          â”‚ P10_DNA_L001_2    â”‚ 2  â”‚\n",
      "â”‚ P105         â”‚ P105_DNA_L001_1   â”‚ 2  â”‚\n",
      "â”‚ P113         â”‚ P113_DNA_L001     â”‚ 2  â”‚\n",
      "â”‚ P12          â”‚ P12_DNA_L001_2    â”‚ 2  â”‚\n",
      "â”‚ P121         â”‚ P121_DNA_L001     â”‚ 2  â”‚\n",
      "â”‚ P122         â”‚ P122_DNA_L001     â”‚ 2  â”‚\n",
      "â”‚ P13          â”‚ P13_DNA_L001_2    â”‚ 2  â”‚\n",
      "â”‚ P133         â”‚ P133_DNA_L001     â”‚ 2  â”‚\n",
      "â”‚ P138         â”‚ P138_DNA_L001     â”‚ 2  â”‚\n",
      "â””â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”˜\n"
     ]
    }
   ],
   "source": [
    "sqlite3 -batch sample_collab.db \\\n",
    "\"select patient_code,host_subject_id,count(*) as nb \\\n",
    "from ((SraRunTable sra inner join SampleOverviews spl on sra.host_subject_id=spl.sample_name) \\\n",
    "inner join ENARunTable ena on sra.Run=ena.run_accession) \\\n",
    "left join CtValues ct on patient_code=ct.Blinded_number \\\n",
    "group by patient_code order by nb desc limit 20;\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "606bb665-4efc-406b-becf-d1ee2226d3d8",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "â”Œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”\n",
      "â”‚  host_subject_id  â”‚\n",
      "â”œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¤\n",
      "â”‚ P011_DNA_S39_L001 â”‚\n",
      "â”‚ P11_DNA_L001_2    â”‚\n",
      "â”‚ P011_RNA_S43_L001 â”‚\n",
      "â”‚ P11_CDNA_L001_2   â”‚\n",
      "â””â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”˜\n"
     ]
    }
   ],
   "source": [
    "sqlite3 -batch sample_collab.db \\\n",
    "\"select host_subject_id \\\n",
    "from ((SraRunTable sra inner join SampleOverviews spl on sra.host_subject_id=spl.sample_name) \\\n",
    "inner join ENARunTable ena on sra.Run=ena.run_accession) \\\n",
    "left join CtValues ct on patient_code=ct.Blinded_number \\\n",
    "where patient_code='P11';\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "42921cb1-7ef7-4772-81f7-28927778cb6e",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "â”Œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”\n",
      "â”‚  host_subject_id  â”‚ patient_code â”‚ nuc â”‚ total_reads â”‚ nonhost_reads â”‚   Bases   â”‚ host_body_site â”‚ host_disease_status â”‚   miscellaneous_parameter   â”‚  Ct   â”‚    sample_name    â”‚ read_count â”‚ base_count â”‚\n",
      "â”œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¤\n",
      "â”‚ NP_A_DNA_L001     â”‚ NP_A         â”‚ DNA â”‚ 26613104    â”‚ 191868        â”‚ 21290514  â”‚ Nasopharyngeal â”‚ SARS-CoV-2 negative â”‚ Asymptomatic-DNA-sequenced  â”‚       â”‚ NP_A_DNA_L001     â”‚ 191868     â”‚ 21290514   â”‚\n",
      "â”‚ NP_A_cDNA_L001    â”‚ NP_A         â”‚ RNA â”‚ 21402054    â”‚ 290140        â”‚ 32101115  â”‚ Nasopharyngeal â”‚ SARS-CoV-2 negative â”‚ Asymptomatic-cDNA-sequenced â”‚       â”‚ NP_A_cDNA_L001    â”‚ 290140     â”‚ 32101115   â”‚\n",
      "â”‚ NP_B_DNA_L001     â”‚ NP_B         â”‚ DNA â”‚ 19924574    â”‚ 399904        â”‚ 42664989  â”‚ Nasopharyngeal â”‚ SARS-CoV-2 negative â”‚ Asymptomatic-DNA-sequenced  â”‚       â”‚ NP_B_DNA_L001     â”‚ 399904     â”‚ 42664989   â”‚\n",
      "â”‚ NP_B_cDNA_L001    â”‚ NP_B         â”‚ RNA â”‚ 23805264    â”‚ 1309086       â”‚ 152208469 â”‚ Nasopharyngeal â”‚ SARS-CoV-2 negative â”‚ Asymptomatic-cDNA-sequenced â”‚       â”‚ NP_B_cDNA_L001    â”‚ 1309086    â”‚ 152208469  â”‚\n",
      "â”‚ NP_C_DNA_L001     â”‚ NP_C         â”‚ DNA â”‚ 16834116    â”‚ 1449880       â”‚ 157565410 â”‚ Nasopharyngeal â”‚ SARS-CoV-2 negative â”‚ Asymptomatic-DNA-sequenced  â”‚       â”‚ NP_C_DNA_L001     â”‚ 1449880    â”‚ 157565410  â”‚\n",
      "â”‚ NP_C_cDNA_L001    â”‚ NP_C         â”‚ RNA â”‚ 20576622    â”‚ 106838        â”‚ 12403605  â”‚ Nasopharyngeal â”‚ SARS-CoV-2 negative â”‚ Asymptomatic-cDNA-sequenced â”‚       â”‚ NP_C_cDNA_L001    â”‚ 106838     â”‚ 12403605   â”‚\n",
      "â”‚ NP_D_DNA_L001     â”‚ NP_D         â”‚ DNA â”‚ 20282170    â”‚ 1999270       â”‚ 226247606 â”‚ Nasopharyngeal â”‚ SARS-CoV-2 negative â”‚ Asymptomatic-DNA-sequenced  â”‚       â”‚ NP_D_DNA_L001     â”‚ 1999270    â”‚ 226247606  â”‚\n",
      "â”‚ NP_D_cDNA_L001    â”‚ NP_D         â”‚ RNA â”‚ 18121696    â”‚ 23678         â”‚ 2638594   â”‚ Nasopharyngeal â”‚ SARS-CoV-2 negative â”‚ Asymptomatic-cDNA-sequenced â”‚       â”‚ NP_D_cDNA_L001    â”‚ 23678      â”‚ 2638594    â”‚\n",
      "â”‚ NP_E_DNA_L001     â”‚ NP_E         â”‚ DNA â”‚ 17098832    â”‚ 1060832       â”‚ 114446653 â”‚ Nasopharyngeal â”‚ SARS-CoV-2 negative â”‚ Asymptomatic-DNA-sequenced  â”‚       â”‚ NP_E_DNA_L001     â”‚ 1060832    â”‚ 114446653  â”‚\n",
      "â”‚ NP_E_cDNA_L001    â”‚ NP_E         â”‚ RNA â”‚ 16381626    â”‚ 47250         â”‚ 5524154   â”‚ Nasopharyngeal â”‚ SARS-CoV-2 negative â”‚ Asymptomatic-cDNA-sequenced â”‚       â”‚ NP_E_cDNA_L001    â”‚ 47250      â”‚ 5524154    â”‚\n",
      "â”‚ P004_DNA_L001     â”‚ P4           â”‚ DNA â”‚ 20246544    â”‚ 658268        â”‚ 80692720  â”‚ Nasopharyngeal â”‚ SARS-CoV-2 positive â”‚ Symptomatic-DNA-sequenced   â”‚ 18.0  â”‚ P004_DNA_L001     â”‚ 658268     â”‚ 80692720   â”‚\n",
      "â”‚ P004_cDNA_L001    â”‚ P4           â”‚ RNA â”‚ 27027782    â”‚ 758080        â”‚ 89552300  â”‚ Nasopharyngeal â”‚ SARS-CoV-2 positive â”‚ Symptomatic-cDNA-sequenced  â”‚ 18.0  â”‚ P004_cDNA_L001    â”‚ 758080     â”‚ 89552300   â”‚\n",
      "â”‚ P011_DNA_S39_L001 â”‚ P11          â”‚ DNA â”‚ 27630988    â”‚ 102524        â”‚ 13193813  â”‚ Nasopharyngeal â”‚ SARS-CoV-2 positive â”‚ Symptomatic-DNA-sequenced   â”‚ 23.59 â”‚ P011_DNA_S39_L001 â”‚ 102524     â”‚ 13193813   â”‚\n",
      "â”‚ P011_RNA_S43_L001 â”‚ P11          â”‚ RNA â”‚ 17500872    â”‚ 9602          â”‚ 1171941   â”‚ Nasopharyngeal â”‚ SARS-CoV-2 positive â”‚ Symptomatic-cDNA-sequenced  â”‚ 23.59 â”‚ P011_RNA_S43_L001 â”‚ 9602       â”‚ 1171941    â”‚\n",
      "â”‚ P019_DNA_S74_L001 â”‚ P19          â”‚ DNA â”‚ 21364260    â”‚ 385740        â”‚ 49060845  â”‚ Nasopharyngeal â”‚ SARS-CoV-2 positive â”‚ Symptomatic-DNA-sequenced   â”‚ 18.81 â”‚ P019_DNA_S74_L001 â”‚ 385740     â”‚ 49060845   â”‚\n",
      "â”‚ P019_RNA_S78_L001 â”‚ P19          â”‚ RNA â”‚ 20758064    â”‚ 62396         â”‚ 6673367   â”‚ Nasopharyngeal â”‚ SARS-CoV-2 positive â”‚ Symptomatic-cDNA-sequenced  â”‚ 18.81 â”‚ P019_RNA_S78_L001 â”‚ 62396      â”‚ 6673367    â”‚\n",
      "â”‚ P01_CDNA_L001_2   â”‚ P1           â”‚ RNA â”‚ 20429462    â”‚ 606162        â”‚ 80609874  â”‚ Nasopharyngeal â”‚ SARS-CoV-2 negative â”‚ Asymptomatic-cDNA-sequenced â”‚ 27.65 â”‚ P01_CDNA_L001_2   â”‚ 606162     â”‚ 80609874   â”‚\n",
      "â”‚ P01_DNA_L001_2    â”‚ P1           â”‚ DNA â”‚ 17377298    â”‚ 1769196       â”‚ 248757369 â”‚ Nasopharyngeal â”‚ SARS-CoV-2 negative â”‚ Asymptomatic-DNA-sequenced  â”‚ 27.65 â”‚ P01_DNA_L001_2    â”‚ 1769196    â”‚ 248757369  â”‚\n",
      "â”‚ P022_DNA_L001     â”‚ P22          â”‚ DNA â”‚ 22910206    â”‚ 610606        â”‚ 76430173  â”‚ Nasopharyngeal â”‚ SARS-CoV-2 positive â”‚ Symptomatic-DNA-sequenced   â”‚ 18.02 â”‚ P022_DNA_L001     â”‚ 610606     â”‚ 76430173   â”‚\n",
      "â”‚ P022_cDNA_L001    â”‚ P22          â”‚ RNA â”‚ 19148958    â”‚ 1392334       â”‚ 184178409 â”‚ Nasopharyngeal â”‚ SARS-CoV-2 positive â”‚ Symptomatic-cDNA-sequenced  â”‚ 18.02 â”‚ P022_cDNA_L001    â”‚ 1392334    â”‚ 184178409  â”‚\n",
      "â””â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”˜\n"
     ]
    }
   ],
   "source": [
    "sqlite3 -batch sample_collab.db \\\n",
    "\"select host_subject_id,patient_code,nucleotide_type as nuc,total_reads,nonhost_reads,Bases,host_body_site,\\\n",
    "    host_disease_status,miscellaneous_parameter,Ct, spl.sample_name, read_count,base_count \\\n",
    "from ((SraRunTable sra inner join SampleOverviews spl on sra.host_subject_id=spl.sample_name) \\\n",
    "inner join ENARunTable ena on sra.Run=ena.run_accession) \\\n",
    "left join CtValues ct on patient_code=ct.Blinded_number \\\n",
    "order by host_subject_id asc limit 20;\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "09de22d3-a8d8-4d4b-b129-5acac095c69b",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "â”Œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”\n",
      "â”‚ count(*) â”‚\n",
      "â”œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¤\n",
      "â”‚ 134      â”‚\n",
      "â””â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”˜\n"
     ]
    }
   ],
   "source": [
    "sqlite3 -batch sample_collab.db \\\n",
    "\"select count(*) \\\n",
    "from ((SraRunTable sra inner join SampleOverviews spl on sra.host_subject_id=spl.sample_name) \\\n",
    "inner join ENARunTable ena on sra.Run=ena.run_accession) \\\n",
    "left join CtValues ct on patient_code=ct.Blinded_number \\\n",
    "where Ct>0;\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "9936f7e2-b249-48bd-8427-f66d7be4294c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "â”Œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”\n",
      "â”‚ username â”‚ firstname â”‚ lastname â”‚\n",
      "â”œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¤\n",
      "â”‚ phingamp â”‚ Pascal    â”‚ Hingamp  â”‚\n",
      "â””â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”˜\n"
     ]
    }
   ],
   "source": [
    "sqlite3 -batch sample_collab.db \"select * from bioinformaticians;\"\n",
    "sqlite3 -batch sample_collab.db \"insert into bioinformaticians (username,firstname,lastname) values ('phingamp','Pascal','Hingamp');\"\n",
    "sqlite3 -batch sample_collab.db \"select * from bioinformaticians;\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "1fdb6c17-eb1f-45f3-bb53-5d62c5600ce2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "â”Œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”\n",
      "â”‚ username â”‚ patient_code â”‚\n",
      "â”œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¤\n",
      "â”‚ phingamp â”‚ NP_A         â”‚\n",
      "â”‚ phingamp â”‚ NP_B         â”‚\n",
      "â”‚ phingamp â”‚ NP_C         â”‚\n",
      "â”‚ phingamp â”‚ NP_D         â”‚\n",
      "â”‚ phingamp â”‚ NP_E         â”‚\n",
      "â””â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”˜\n"
     ]
    }
   ],
   "source": [
    "sqlite3 -batch sample_collab.db \"select * from sample2bioinformatician;\"\n",
    "sqlite3 -batch sample_collab.db \"insert into sample2bioinformatician (username,patient_code) values ('phingamp','NP_A');\"\n",
    "sqlite3 -batch sample_collab.db \"insert into sample2bioinformatician (username,patient_code) values ('phingamp','NP_B');\"\n",
    "sqlite3 -batch sample_collab.db \"insert into sample2bioinformatician (username,patient_code) values ('phingamp','NP_C');\"\n",
    "sqlite3 -batch sample_collab.db \"insert into sample2bioinformatician (username,patient_code) values ('phingamp','NP_D');\"\n",
    "sqlite3 -batch sample_collab.db \"insert into sample2bioinformatician (username,patient_code) values ('phingamp','NP_E');\"\n",
    "sqlite3 -batch sample_collab.db \"select * from sample2bioinformatician;\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "cca9d983-4659-4f10-80d1-a5db5f6203e3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "â”Œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”\n",
      "â”‚ run_accession â”‚  host_subject_id  â”‚ patient_code â”‚ nuc â”‚ host_body_site â”‚ host_disease_status â”‚   miscellaneous_parameter   â”‚  Ct   â”‚ total_reads â”‚ read_count â”‚ base_count â”‚ username â”‚ firstname â”‚ lastname â”‚\n",
      "â”œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¤\n",
      "â”‚ ERR6913174    â”‚ NP_A_DNA_L001     â”‚ NP_A         â”‚ DNA â”‚ Nasopharyngeal â”‚ SARS-CoV-2 negative â”‚ Asymptomatic-DNA-sequenced  â”‚       â”‚ 26613104    â”‚ 191868     â”‚ 21290514   â”‚ phingamp â”‚ Pascal    â”‚ Hingamp  â”‚\n",
      "â”‚ ERR6913175    â”‚ NP_B_DNA_L001     â”‚ NP_B         â”‚ DNA â”‚ Nasopharyngeal â”‚ SARS-CoV-2 negative â”‚ Asymptomatic-DNA-sequenced  â”‚       â”‚ 19924574    â”‚ 399904     â”‚ 42664989   â”‚ phingamp â”‚ Pascal    â”‚ Hingamp  â”‚\n",
      "â”‚ ERR6913270    â”‚ NP_A_cDNA_L001    â”‚ NP_A         â”‚ RNA â”‚ Nasopharyngeal â”‚ SARS-CoV-2 negative â”‚ Asymptomatic-cDNA-sequenced â”‚       â”‚ 21402054    â”‚ 290140     â”‚ 32101115   â”‚ phingamp â”‚ Pascal    â”‚ Hingamp  â”‚\n",
      "â”‚ ERR6913177    â”‚ NP_D_DNA_L001     â”‚ NP_D         â”‚ DNA â”‚ Nasopharyngeal â”‚ SARS-CoV-2 negative â”‚ Asymptomatic-DNA-sequenced  â”‚       â”‚ 20282170    â”‚ 1999270    â”‚ 226247606  â”‚ phingamp â”‚ Pascal    â”‚ Hingamp  â”‚\n",
      "â”‚ ERR6913178    â”‚ NP_E_DNA_L001     â”‚ NP_E         â”‚ DNA â”‚ Nasopharyngeal â”‚ SARS-CoV-2 negative â”‚ Asymptomatic-DNA-sequenced  â”‚       â”‚ 17098832    â”‚ 1060832    â”‚ 114446653  â”‚ phingamp â”‚ Pascal    â”‚ Hingamp  â”‚\n",
      "â”‚ ERR6913273    â”‚ NP_D_cDNA_L001    â”‚ NP_D         â”‚ RNA â”‚ Nasopharyngeal â”‚ SARS-CoV-2 negative â”‚ Asymptomatic-cDNA-sequenced â”‚       â”‚ 18121696    â”‚ 23678      â”‚ 2638594    â”‚ phingamp â”‚ Pascal    â”‚ Hingamp  â”‚\n",
      "â”‚ ERR6913274    â”‚ NP_E_cDNA_L001    â”‚ NP_E         â”‚ RNA â”‚ Nasopharyngeal â”‚ SARS-CoV-2 negative â”‚ Asymptomatic-cDNA-sequenced â”‚       â”‚ 16381626    â”‚ 47250      â”‚ 5524154    â”‚ phingamp â”‚ Pascal    â”‚ Hingamp  â”‚\n",
      "â”‚ ERR6913176    â”‚ NP_C_DNA_L001     â”‚ NP_C         â”‚ DNA â”‚ Nasopharyngeal â”‚ SARS-CoV-2 negative â”‚ Asymptomatic-DNA-sequenced  â”‚       â”‚ 16834116    â”‚ 1449880    â”‚ 157565410  â”‚ phingamp â”‚ Pascal    â”‚ Hingamp  â”‚\n",
      "â”‚ ERR6913272    â”‚ NP_C_cDNA_L001    â”‚ NP_C         â”‚ RNA â”‚ Nasopharyngeal â”‚ SARS-CoV-2 negative â”‚ Asymptomatic-cDNA-sequenced â”‚       â”‚ 20576622    â”‚ 106838     â”‚ 12403605   â”‚ phingamp â”‚ Pascal    â”‚ Hingamp  â”‚\n",
      "â”‚ ERR6913271    â”‚ NP_B_cDNA_L001    â”‚ NP_B         â”‚ RNA â”‚ Nasopharyngeal â”‚ SARS-CoV-2 negative â”‚ Asymptomatic-cDNA-sequenced â”‚       â”‚ 23805264    â”‚ 1309086    â”‚ 152208469  â”‚ phingamp â”‚ Pascal    â”‚ Hingamp  â”‚\n",
      "â”‚ ERR6913102    â”‚ P011_DNA_S39_L001 â”‚ P11          â”‚ DNA â”‚ Nasopharyngeal â”‚ SARS-CoV-2 positive â”‚ Symptomatic-DNA-sequenced   â”‚ 23.59 â”‚ 27630988    â”‚ 102524     â”‚ 13193813   â”‚          â”‚           â”‚          â”‚\n",
      "â”‚ ERR6913103    â”‚ P019_DNA_S74_L001 â”‚ P19          â”‚ DNA â”‚ Nasopharyngeal â”‚ SARS-CoV-2 positive â”‚ Symptomatic-DNA-sequenced   â”‚ 18.81 â”‚ 21364260    â”‚ 385740     â”‚ 49060845   â”‚          â”‚           â”‚          â”‚\n",
      "â”‚ ERR6913108    â”‚ P069_DNA_S27_L001 â”‚ P69          â”‚ DNA â”‚ Nasopharyngeal â”‚ SARS-CoV-2 positive â”‚ Symptomatic-DNA-sequenced   â”‚ 18.42 â”‚ 23356096    â”‚ 226280     â”‚ 29240147   â”‚          â”‚           â”‚          â”‚\n",
      "â”‚ ERR6913111    â”‚ P090_DNA_S25_L001 â”‚ P90          â”‚ DNA â”‚ Nasopharyngeal â”‚ SARS-CoV-2 positive â”‚ Symptomatic-DNA-sequenced   â”‚ 26.98 â”‚ 27646936    â”‚ 33240      â”‚ 4397612    â”‚          â”‚           â”‚          â”‚\n",
      "â”‚ ERR6913115    â”‚ P267_DNA_S9_L001  â”‚ P267         â”‚ DNA â”‚ Nasopharyngeal â”‚ SARS-CoV-2 positive â”‚ Symptomatic-DNA-sequenced   â”‚ 19.63 â”‚ 18306476    â”‚ 33254      â”‚ 4259170    â”‚          â”‚           â”‚          â”‚\n",
      "â”‚ ERR6913116    â”‚ P269_DNA_S3_L001  â”‚ P269         â”‚ DNA â”‚ Nasopharyngeal â”‚ SARS-CoV-2 positive â”‚ Symptomatic-DNA-sequenced   â”‚ 23.43 â”‚ 21061004    â”‚ 43790      â”‚ 5716941    â”‚          â”‚           â”‚          â”‚\n",
      "â”‚ ERR6913117    â”‚ P272_DNA_S40_L001 â”‚ P272         â”‚ DNA â”‚ Nasopharyngeal â”‚ SARS-CoV-2 positive â”‚ Symptomatic-DNA-sequenced   â”‚ 21.34 â”‚ 24205196    â”‚ 397186     â”‚ 50556736   â”‚          â”‚           â”‚          â”‚\n",
      "â”‚ ERR6913118    â”‚ P273_DNA_L001     â”‚ P273         â”‚ DNA â”‚ Nasopharyngeal â”‚ SARS-CoV-2 positive â”‚ Symptomatic-DNA-sequenced   â”‚ 17.81 â”‚ 16700384    â”‚ 140270     â”‚ 16091403   â”‚          â”‚           â”‚          â”‚\n",
      "â”‚ ERR6913136    â”‚ P27_DNA_L001_2    â”‚ P27          â”‚ DNA â”‚ Nasopharyngeal â”‚ SARS-CoV-2 positive â”‚ Asymptomatic-DNA-sequenced  â”‚ 33.4  â”‚ 20026148    â”‚ 2239760    â”‚ 318332989  â”‚          â”‚           â”‚          â”‚\n",
      "â”‚ ERR6913147    â”‚ P38_DNA_L001_2    â”‚ P38          â”‚ DNA â”‚ Nasopharyngeal â”‚ SARS-CoV-2 positive â”‚ Asymptomatic-DNA-sequenced  â”‚ 29.33 â”‚ 18680552    â”‚ 1994778    â”‚ 274525044  â”‚          â”‚           â”‚          â”‚\n",
      "â””â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”˜\n"
     ]
    }
   ],
   "source": [
    "sqlite3 -batch sample_collab.db \\\n",
    "\"select * from sample_annot spl left join (sample2bioinformatician s2b left join bioinformaticians b using(username)) using(patient_code) order by username desc limit 20;\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "a747cc4b-0cd7-4038-8cc7-59b1cf44048a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "â”Œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”\n",
      "â”‚ count(*) â”‚\n",
      "â”œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¤\n",
      "â”‚ 10       â”‚\n",
      "â””â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”˜\n"
     ]
    }
   ],
   "source": [
    "sqlite3 -batch sample_collab.db \\\n",
    "\"select count(*) from sample_annot spl left join (sample2bioinformatician s2b left join bioinformaticians b using(username)) using(patient_code) where username='phingamp';\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0861f0ca-aa6e-4bcf-a036-5637407cf7ce",
   "metadata": {},
   "source": [
    "# Don't forget to commit changes and leave this sandbox branch !\n",
    "\n",
    "Change in the last git command below the branch name to your personal development branch name (should be your firstname)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dafe34fe-699e-4237-9c3e-4c529fa07008",
   "metadata": {},
   "outputs": [],
   "source": [
    "git add .\n",
    "git commit -m \"completed sandbox DB build\"\n",
    "git checkout ==your_firstname=="
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
